tests/System.Text.JsonLab.Tests/JsonReaderTests.cs

```diff
@@ -308,22 +308,141 @@ public static void TestJsonReaderUtf8SpecialString(string jsonString, string exp
             Assert.Equal(expectedStr, actualStr);
         }
 
+
+        // TODO: Need a mechanism to pass state appropriately
+        [Theory]
+        [InlineData("\"hello\"", 1, 0)] // "\""
+        [InlineData("12345", 3, 0)]   // "123"
+        [InlineData("null", 3, 0)]   // "nul"
+        [InlineData("true", 3, 0)]   // "tru"
+        [InlineData("false", 4, 0)]  // "fals"
+        //[InlineData("{\"age\":30}", 7, 7)] // "{\"age\":"
+        //[InlineData("{\"name\":\"Ahson\"", 9, 8)]  // "{\"name\":\"Ahso"
+        [InlineData("-123456789", 1, 0)] // "-"
+        [InlineData("0.5", 2, 0)]    // "0."
+        [InlineData("10.5e+3", 5, 0)] // "10.5e"
+        [InlineData("10.5e-1", 6, 0)]    // "10.5e-"
+        //[InlineData("{\"ints\":[1, 2, 3, 4, 5]}", 21, 21)]    // "{\"ints\":[1, 2, 3, 4, 5"
+        //[InlineData("{\"strings\":[\"abc\", \"def\"], \"ints\":[1, 2, 3, 4, 5]}", 24, 24)]  // "{\"strings\":[\"abc\", \"def\""
+        //[InlineData("{\"age\":30, \"name\":\"test}:[]\", \"another string\" : \"tests\"}", 19, 18)]   // "{\"age\":30, \"name\":\"test}"
+        public static void SplitJsonAsync(string jsonString, int splitLocation, int consumed)
+        {
+            var buffer = new char[jsonString.Length];
+            var reader = new StringReader(jsonString);
+
+            reader.ReadAsync(buffer, 0, splitLocation);
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(buffer, 0, splitLocation);
+            var json = new Utf8JsonReader(dataUtf8, false);
+            while (json.Read()) ;
+
+            Assert.Equal(consumed, json.CurrentIndex);
+
+            reader.ReadAsync(buffer, splitLocation, jsonString.Length - splitLocation);
+
+            int start = splitLocation - (dataUtf8.Length - json.CurrentIndex);
+            dataUtf8 = Encoding.UTF8.GetBytes(buffer, start, jsonString.Length - start);
+            json = new Utf8JsonReader(dataUtf8, true, json.TokenType);
+            while (json.Read()) ;
+
+            Assert.Equal(dataUtf8.Length, json.CurrentIndex);
+        }
+
+        [Theory]
+        [InlineData("\"", 0)]
+        [InlineData("123", 0)]
+        [InlineData("nul", 0)]
+        [InlineData("tru", 0)]
+        [InlineData("fals", 0)]
+        [InlineData("{\"age\":", 7)]
+        [InlineData("{\"name\":\"Ahso", 8)]
+        [InlineData("-", 0)]
+        [InlineData("0.", 0)]
+        [InlineData("10.5e", 0)]
+        [InlineData("10.5e-", 0)]
+        [InlineData("{\"ints\":[1, 2, 3, 4, 5", 21)]
+        [InlineData("{\"strings\":[\"abc\", \"def\"", 24)]
+        [InlineData("{\"age\":30, \"name\":\"test}", 18)]
+        [InlineData("[[[[{\r\n\"temp1\":[[[[{\"temp2:[]}]]]]}]]]]", 20)]
+        public static void PartialJson(string jsonString, int consumed)
```

Comment:
Does this test anything beyond what the tests above does?

Reply:
Good point, not it doesn't.